docs/source/spec/aws-json-protocols.rst

```diff
@@ -0,0 +1,103 @@
+.. _aws-json-protocols:
```

Comment:
The protocol docs are very incomplete at this point. We need to circle back to them and fill in the details on how they work. This can be based on the accompanying protocol tests: https://github.com/awslabs/smithy/tree/master/smithy-aws-protocol-tests/model